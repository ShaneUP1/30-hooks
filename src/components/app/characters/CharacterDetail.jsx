import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ character }) => {
  const episodeList = character.episode.map((episode) => (
    <li key={episode}>{episode}</li>
  ));
  
  return (
    <figure data-testid="detail">
      <img src={character.image} alt={character.name} />
      <figcaption>
        <h2>{character.name}</h2>
        <p>{character.species}</p>
        <p>{character.origin.name}</p>
        <ul>
          {episodeList}
        </ul>
      </figcaption>
    </figure>
  );
};


CharacterDetail.propTypes = {
  character: PropTypes.object.isRequired
};

export default CharacterDetail;
