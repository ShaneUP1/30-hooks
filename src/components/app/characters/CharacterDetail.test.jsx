/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';


describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders a detail of a character', () => {
    const { asFragment } = render(<CharacterDetail
      character={{ name: 'shane', image: 'www.url.com', species: 'toad', id: 1, origin:{ name: 'earth' }, episode: ['https://rickandmortyapi.com/api/episode/1'] }}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
