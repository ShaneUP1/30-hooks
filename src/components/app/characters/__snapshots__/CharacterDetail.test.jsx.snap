// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders a detail of a character 1`] = `
<DocumentFragment>
  <figure
    data-testid="detail"
  >
    <img
      alt="shane"
      src="www.url.com"
    />
    <figcaption>
      <h2>
        shane
      </h2>
      <p>
        toad
      </p>
      <p>
        earth
      </p>
      <ul>
        <li>
          https://rickandmortyapi.com/api/episode/1
        </li>
      </ul>
    </figcaption>
  </figure>
</DocumentFragment>
`;
